components/features/TemplateSection.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <div class="container py-5">
+    <div class="row px-0">
+      <div class="heading d-flex flex-column align-items-center text-center">
+        <SectionTitle :heading="title" />
+        <p class="hero__subheading" v-if="description">
+          {{ description }}
+        </p>
+      </div>
+      <div v-if="templates && templates.length" class="templates py-3">
+        <TemplateCard
+          v-for="(template, idx) in templates"
+          :key="idx"
+          :template="template"
+        />
+      </div>
+      <div class="d-flex align-items-center justify-content-center mt-4">
+        <NuxtLink :to="button.link">
+          <button class="btn-all-templates">{{button.text}}</button>
+        </NuxtLink>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import TemplateCard from '@/components/template/TemplateCard.vue'
+import axios from 'axios'
+
+export default {
+  components: { TemplateCard },
+  props: {
+    slug: {
+      type: String,
+      default: '',
+    },
+    title: {
+      type: Array,
+      required: true,
+    },
+    description: {
+      type: String,
+      default: '',
+    },
+    button: {
+      type: Object,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      templates: [],
+    }
+  },
+  mounted() {
+    this.getTemplates()
+  },
+  methods: {
+    async getTemplates() {
+      try {
+        let response
+        if (this.slug) {
+          response = await axios.get(
+            'https://app.formester.com/templates.json',
+            {
+              params: { category_slug: this.slug },
+            }
+          )
+        } else {
+          response = await axios.get('https://app.formester.com/templates.json')
+        }
+        let templates = response.data
+
+        const dummyDescription =
+          'Check out this pre-designed template and start customising with just a single click. Personalise with your branding, incorporate electronic signatures for security and add multiple collaborators to make changes simultaneously. Use this template and start getting data driven actionable insights with robust analytics.'
+
+        templates = templates.map((template) => ({
+          ...template,
+          description: template.description || dummyDescription,
+        }))
+
+        const randIndex = Math.floor(Math.random() * (templates.length - 3))
+        this.templates = templates.slice(randIndex, randIndex + 3)
+      } catch (err) {
+        console.error(err)
+      }
```

Comment:
Does this fetch all the templates just to render 3 random templates every time?

Reply:
Yes. It was done like this before. Is there any other route for getting 3 templates?

Reply:
No sure. But we can update our backend to do it from backend. Using params.
eg: `https://app.formester.com/templates.json?sort=random&limit=3`

Reply:
I got it.. I'll do this.